src/lib/transition_frontier_controller_tests/test_catchup_scheduler.ml

```diff
@@ -0,0 +1,400 @@
+          let dangling_breadcrumb3 = List.nth_exn upcoming_breadcrumbs 3 in
+          let missing_transition =
+            Transition_frontier.Breadcrumb.transition_with_hash
+              missing_breadcrumb
+          in
+          let dangling_transition1 =
+            Transition_frontier.Breadcrumb.transition_with_hash
+              dangling_breadcrumb1
+          in
+          let dangling_transition2 =
+            Transition_frontier.Breadcrumb.transition_with_hash
+              dangling_breadcrumb2
+          in
+          let dangling_transition3 =
+            Transition_frontier.Breadcrumb.transition_with_hash
+              dangling_breadcrumb3
+          in
+          Catchup_scheduler.watch scheduler ~timeout_duration
+            ~transition:dangling_transition3 ;
+          assert (Catchup_scheduler.has_timeout scheduler dangling_transition3) ;
+          Catchup_scheduler.watch scheduler ~timeout_duration
+            ~transition:dangling_transition2 ;
+          assert (
+            Catchup_scheduler.has_timeout scheduler dangling_transition2
+            && not
+               @@ Catchup_scheduler.has_timeout scheduler dangling_transition3
+          ) ;
+          Catchup_scheduler.watch scheduler ~timeout_duration
+            ~transition:dangling_transition1 ;
+          assert (
+            Catchup_scheduler.has_timeout scheduler dangling_transition1
+            && not
+               @@ Catchup_scheduler.has_timeout scheduler dangling_transition2
+          ) ;
+          Transition_frontier.add_breadcrumb_exn frontier missing_breadcrumb ;
+          Catchup_scheduler.notify scheduler ~transition:missing_transition
+          |> ignore ;
+          assert (Catchup_scheduler.is_empty scheduler) ;
+          let result_ivar = Ivar.create () in
+          Strict_pipe.Reader.iter_without_pushback catchup_breadcrumbs_reader
+            ~f:(fun dangling_rose_trees ->
+              Ivar.fill result_ivar dangling_rose_trees )
+          |> don't_wait_for ;
+          let%map dangling_rose_trees = Ivar.read result_ivar in
+          let received_rose_tree =
+            Rose_tree.T (missing_breadcrumb, dangling_rose_trees)
+          in
+          assert (
+            Rose_tree.equal received_rose_tree
+              (Rose_tree.of_list_exn upcoming_breadcrumbs)
+              ~f:(fun breadcrumb1 breadcrumb2 ->
+                External_transition.Verified.equal
+                  (With_hash.data
+                     (Transition_frontier.Breadcrumb.transition_with_hash
+                        breadcrumb1))
+                  (With_hash.data
+                     (Transition_frontier.Breadcrumb.transition_with_hash
+                        breadcrumb2)) ) ) ;
+          Strict_pipe.Writer.close catchup_breadcrumbs_writer ;
+          Strict_pipe.Writer.close catchup_job_writer )
+
+    let%test_unit "If a non-linear sequence of transitions in order and the \
```

Comment:
The test "if a linear sequence of transitions come out of order, catchup scheduler should not create duplicate jobs" is a more specific version of this one. Can you remove the more specific test and for this test, test with linear_gen and k_tree_gen